src/openms/source/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.cpp

```diff
@@ -391,82 +396,93 @@ namespace OpenMS
 
     for (auto const& pep_q : pep_quant_)
     {
-      String accession = getAccession_(pep_q.second.accessions,
-                                       accession_to_leader);
+      String accession = getAccession_(pep_q.second.accessions, accession_to_leader);
       OPENMS_LOG_DEBUG << "Peptide id mapped to leader: " << accession << endl;
 
       // not enough evidence or mapping to multiple groups
-      if (accession.empty()) continue;
+      if (accession.empty())
+        continue;
 
       // proteotypic peptide
       const String peptide = pep_q.first.toUnmodifiedString();
 
       prot_quant_[accession].psm_count += pep_q.second.psm_count;
 
       // transfer abundances and counts from peptides->protein
-      // summarize abundances and counts between different peptidoforms       
-      for (auto const & sta : pep_q.second.total_abundances)
+      // summarize abundances and counts between different peptidoforms
+      for (auto const& sta : pep_q.second.total_abundances)
       {
         prot_quant_[accession].abundances[peptide][sta.first] += sta.second;
       }
 
-      for (auto const & sta : pep_q.second.total_psm_counts)
+      for (auto const& sta : pep_q.second.total_psm_counts)
       {
         prot_quant_[accession].psm_counts[peptide][sta.first] += sta.second;
       }
     }
 
+    std::string method = param_.getValue("method");
+
+    if (method == "iBAQ")  // TODO: parameter override too late - ProteinQuantifier reads parameters earlier, meaning output file contains incorrect parameters
```

Comment:
I think this could be confusing for a developer if after setting to iBAQ, running the algorithm and then setting it to "top" again the param_ object has changed.
I think this needs to be changed.

Reply:
e.g. if top or average is used at some point check there again for the method and use if / else for the two cases